<a href="https://colab.research.google.com/github/umiSirya/General-Data-analysis/blob/main/Housing_Market.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Housing Market

### Introduction:

This time we will create our own dataset with fictional numbers to describe a house market. As we are going to create random data don't try to reason of the numbers.

### Step 1. Import the necessary libraries

In [ ]:
import pandas as pd
import numpy as np

### Step 2. Create 3 differents Series, each of length 100, as follows:
1. The first a random number from 1 to 4
2. The second a random number from 1 to 3
3. The third a random number from 10,000 to 30,000

In [ ]:
series1 = pd.Series(np.random.randint(1, 5, size=100), name="Series1")
series2 = pd.Series(np.random.randint(1, 4, size=100), name="Series2")
series3 = pd.Series(np.random.randint(10000, 30001, size=100), name="Series3")

print(series1.head())
print(series2.head())
print(series3.head())

0    2
1    3
2    2
3    3
4    2
Name: Series1, dtype: int64
0    2
1    2
2    3
3    1
4    3
Name: Series2, dtype: int64
0    23780
1    18350
2    18467
3    20315
4    12398
Name: Series3, dtype: int64


### Step 3. Let's create a DataFrame by joinning the Series by column

In [ ]:
df = pd.concat([series1, series2, series3], axis=1)
df

Series1  Series2  Series3
0         2        2    23780
1         3        2    18350
2         2        3    18467
3         3        1    20315
4         2        3    12398
..      ...      ...      ...
95        3        2    20968
96        2        3    26855
97        1        1    23641
98        4        1    18537
99        4        2    23131

[100 rows x 3 columns]

### Step 4. Change the name of the columns to bedrs, bathrs, price_sqr_meter

In [ ]:
df.rename(columns={0: 'bedrs', 1: 'bathrs', 2: 'price_sqr_meter'}, inplace=True)
df

Series1  Series2  Series3
0         2        2    23780
1         3        2    18350
2         2        3    18467
3         3        1    20315
4         2        3    12398
..      ...      ...      ...
95        3        2    20968
96        2        3    26855
97        1        1    23641
98        4        1    18537
99        4        2    23131

[100 rows x 3 columns]

### Step 5. Create a one column DataFrame with the values of the 3 Series and assign it to 'bigcolumn'

In [ ]:
bigcolumn = pd.concat([series1, series2, series3], axis=0).reset_index(drop=True)

bigcolumn_df = pd.DataFrame(bigcolumn, columns=['bigcolumn'])
bigcolumn_df.head()

bigcolumn
0          2
1          3
2          2
3          3
4          2

### Step 6. Oops, it seems it is going only until index 99. Is it true?

In [10]:
total_rows = bigcolumn_df.shape[0]
last_few_indices = bigcolumn_df.index[-5:]

total_rows, list(last_few_indices)
#yes its true, the indices weren't reset properly after concatenation, so DataFrame might look as if it stops at index 99.

(300, [295, 296, 297, 298, 299])

### Step 7. Reindex the DataFrame so it goes from 0 to 299

In [12]:
bigcolumn_df.reset_index(drop=True, inplace=True)
bigcolumn_df

bigcolumn
0            2
1            3
2            2
3            3
4            2
..         ...
295      20968
296      26855
297      23641
298      18537
299      23131

[300 rows x 1 columns]